lib/internal/modules/esm/loader.js

```diff
@@ -313,28 +369,11 @@ class CustomizedModuleLoader extends DefaultModuleLoader {
    * @returns {{ format: string, url: URL['href'] }}
    */
   resolve(originalSpecifier, parentURL, importAssertions) {
-    return hooksProxy.makeSyncRequest('resolve', originalSpecifier, parentURL, importAssertions);
+    return hooksProxy.makeAsyncRequest('resolve', originalSpecifier, parentURL, importAssertions);
   }
 
-  async #getModuleJob(specifier, parentURL, importAssertions) {
-    const resolveResult = await hooksProxy.makeAsyncRequest('resolve', specifier, parentURL, importAssertions);
-
-    return this.getJobFromResolveResult(resolveResult, parentURL, importAssertions);
-  }
-  getModuleJob(specifier, parentURL, importAssertions) {
-    const jobPromise = this.#getModuleJob(specifier, parentURL, importAssertions);
-
-    return {
-      run() {
-        return PromisePrototypeThen(jobPromise, (job) => job.run());
-      },
-      get modulePromise() {
-        return PromisePrototypeThen(jobPromise, (job) => job.modulePromise);
-      },
-      get linked() {
-        return PromisePrototypeThen(jobPromise, (job) => job.linked);
-      },
-    };
+  resolveSync(originalSpecifier, parentURL, importAssertions) {
+    return hooksProxy.makeSyncRequest('resolve', originalSpecifier, parentURL, importAssertions);
```

Comment:
Can you please explain when `resolveSync` gets called instead of `resolve`? Where’s the logic for choosing one over the other? (And your answer might be good to add as a comment to this method.)

Reply:
Solely for `import.meta.resolve` since it's the one oddball in all this that requires resolution be synchronous – https://github.com/nodejs/node/pull/48559/files#diff-f572ecdefdbdf69b7fc591988665845843bc791a39d116607e27928089f54ed6R19

Reply:
Unresolving cuz i think this should have a code doc (can happen in a follow-up, but the explanation is here).

Reply:
```suggestion
  resolveSync(originalSpecifier, parentURL, importAssertions) {
    // This happens only as a result of `import.meta.resolve` calls, which must be sync per spec.
    return hooksProxy.makeSyncRequest('resolve', originalSpecifier, parentURL, importAssertions);
```